assets/js/payment-methods-demo/express-payment/apple-pay/apple-pay-express.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_STRIPE_EVENT_HANDLERS,
+	PAYMENT_METHOD_NAME,
+} from './constants';
+import {
+	getBillingData,
+	getPaymentMethodData,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptions,
+	normalizeLineItems,
+} from './normalize';
+import {
+	getStripeServerData,
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoApplePay,
+	getTotalPaymentItem,
+	stripePromise,
+} from '../../stripe-utils';
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from '@wordpress/element';
+import {
+	Elements,
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('../../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+// @todo, note it should be trivial to implement cross browser payment api
+// handling here:
+// - ApplePay in Safari
+// - ChromePay in Chrome
+// - not supported for other browsers yet (but if assuming stripe implements the
+//   official PaymentRequest API in their library this should enable support!).
+
+/**
+ * AppleExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const ApplePayExpressComponent = ( {
+	paymentStatus,
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	activePaymentMethod,
+	setActivePaymentMethod,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const stripe = useStripe();
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const eventHandlers = useRef( DEFAULT_STRIPE_EVENT_HANDLERS );
+	const currentBilling = useRef( billing );
+	const currentShipping = useRef( shippingData );
+	const currentPaymentRequest = useRef( paymentRequest );
+	const currentPaymentStatus = useRef( paymentStatus );
+	const currentEventRegistration = useRef( eventRegistration );
+	const isActive = useRef( activePaymentMethod === PAYMENT_METHOD_NAME );
+
+	// update refs when any change
```

Comment:
nit. inline comments should end with full stop - php linters catch this, not sure about js

Reply:
Hmm js linters probably should too... I'm sure I've seen it caught before in the GB project.  I'll fix before merging.